CMakeLists.txt

```
@@ -0,0 +1,263 @@
+
+set(_IS_X86 "\"${_TARGET_ARCH}\" STREQUAL \"x86\" OR \"${_TARGET_ARCH}\" STREQUAL \"x64\"")
+cmake_dependent_option(BASE64_WITH_SSSE3 "add SSSE 3 codepath" ON ${_IS_X86} OFF)
+add_feature_info(SSSE3 BASE64_WITH_SSSE3 "add SSSE 3 codepath")
+cmake_dependent_option(BASE64_WITH_SSE41 "add SSE 4.1 codepath" ON ${_IS_X86} OFF)
+add_feature_info(SSE4.1 BASE64_WITH_SSE41 "add SSE 4.1 codepath")
+cmake_dependent_option(BASE64_WITH_SSE42 "add SSE 4.2 codepath" ON ${_IS_X86} OFF)
+add_feature_info(SSE4.2 BASE64_WITH_SSE42 "add SSE 4.2 codepath")
+cmake_dependent_option(BASE64_WITH_AVX "add AVX codepath" ON ${_IS_X86} OFF)
+add_feature_info(AVX BASE64_WITH_AVX "add AVX codepath")
+cmake_dependent_option(BASE64_WITH_AVX2 "add AVX 2 codepath" ON ${_IS_X86} OFF)
+add_feature_info(AVX2 BASE64_WITH_AVX2 "add AVX2 codepath")
+
+set(_IS_ARM "\"${_TARGET_ARCH}\" STREQUAL \"arm\"")
+cmake_dependent_option(BASE64_WITH_NEON32 "add NEON32 codepath" OFF ${_IS_ARM} OFF)
+add_feature_info(NEON32 BASE64_WITH_NEON32 "add NEON32 codepath")
+
+set(_IS_ARM64 "\"${_TARGET_ARCH}\" STREQUAL \"arm64\"")
+cmake_dependent_option(BASE64_WITH_NEON64 "add NEON64 codepath" ON ${_IS_ARM64} OFF)
+add_feature_info(NEON64 BASE64_WITH_NEON64 "add NEON64 codepath")
+
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/bin")
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/bin")
+
+########################################################################
+# Regenerate headers
+
+if (BASE64_REGENERATE_TABLES)
+    # Generate tables in build folder and copy to source tree.
+    # Don't add the tables in the source tree to the outputs, to avoid `make clean` removing them.
+    add_executable(table_generator
+        lib/tables/table_generator.c
+    )
+
+    add_custom_command(OUTPUT table_dec_32bit.h
+        COMMAND table_generator > table_dec_32bit.h
+        COMMAND "${CMAKE_COMMAND}" -E copy table_dec_32bit.h "${CMAKE_CURRENT_SOURCE_DIR}/lib/tables/table_dec_32bit.h"
+        DEPENDS table_generator
+    )
+    set(Python_ADDITIONAL_VERSIONS 3)
+    find_package(PythonInterp REQUIRED)
+    add_custom_command(OUTPUT table_enc_12bit.h
+        COMMAND "${PYTHON_EXECUTABLE}" "${CMAKE_CURRENT_SOURCE_DIR}/lib/tables/table_enc_12bit.py" > table_enc_12bit.h
+        COMMAND "${CMAKE_COMMAND}" -E copy table_enc_12bit.h "${CMAKE_CURRENT_SOURCE_DIR}/lib/tables/table_enc_12bit.h"
+        DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/lib/tables/table_enc_12bit.py"
+    )
+endif()
+
+
+########################################################################
+# library project
+add_library(base64
+    # library files
+    lib/lib.c
+    lib/codec_choose.c
+    include/libbase64.h
+
+    lib/tables/tables.c
+
+    # codec implementations
+    lib/arch/generic/codec.c
+
+    lib/arch/ssse3/codec.c
+    lib/arch/sse41/codec.c
+    lib/arch/sse42/codec.c
+    lib/arch/avx/codec.c
+    lib/arch/avx2/codec.c
+
+    lib/arch/neon32/codec.c
+    lib/arch/neon64/codec.c
```

You need to add generated sources explicitly to a target.
Otherwise, the `add_custom_command` will never fire.
```suggestion
    lib/arch/neon64/codec.c

    # Add generated headers explicitly to target, to insert them in the dependency tree
    lib/tables/table_dec_32bit.h
    lib/tables/table_enc_12bit.h
```